### DISCUSSION TITLE
GET query parameters

### DISCUSSION THREAD

**Discussion by woodhull** [ElMassimo/js_from_routes#30] [Q&A] [answered]
I'd like to pass in a parameter that should not be interpolated into the URL, but instead included as a GET request query parameter. For example:

```
  await PeopleApi.lookup({params: {account_id: '1234', email: 'foo@bar.com'})
```

The account_id is being interpolated correctly, but I expected the resulting url for the GET request to look like `/accounts/1235/people/lookup?email=foo@bar.com`

Instead, I get: `/accounts/1235/people/lookup`

I've also tried the `data` attribute, but it seems this is not supported for GET requests. 

Is something like this possible (url parameters that are not in the route being appended as query parameters)?

Nathan

**Comment by ElMassimo** [comment, marked as answer]
Hi Nathan!

The behavior is intended to allow passing a full object without sending unnecessary data.

To provide query parameters, do it explicitly [using the `query` option](https://github.com/ElMassimo/js_from_routes/blob/main/packages/core/src/urls.ts#L114):

```ts
  await PeopleApi.lookup({ account_id: '1234', query: { email: 'foo@bar.com' } })
```

Notice that you can safely [omit `params`](https://github.com/ElMassimo/js_from_routes/blob/main/packages/client/src/api.ts#L50) in most cases.
